lib/runner/androidEmulator.js

```diff
@@ -0,0 +1,74 @@
+const untildify = require('untildify');
+const {getPlatformName, getAlreadyRunningAvd, launchAVD, killEmulatorWithoutWait} = require('nightwatch-mobile-helper');
+const {Logger} = require('../utils');
+const path = require('path');
+
+const getSdkRootFromEnv = () => {
+  const androidHome = process.env.ANDROID_HOME;
+
+  if (androidHome) {
+    const androidHomeFinal = untildify(androidHome);
+
+    const androidHomeAbsolute = path.resolve(__dirname, androidHomeFinal);
+    if (androidHomeFinal !== androidHomeAbsolute) {
+      return androidHomeAbsolute;
+    }
+
+    return androidHomeFinal;
+  }
+
+  let err;
+
+  if (androidHome === undefined) {
+    err = new Error('ANDROID_HOME environment variable is NOT set!');
+  } else {
+    err = new Error(`ANDROID_HOME is set to '${androidHome}' which is NOT a valid path!`);
+  }
+  
+  err.help = [
+    `Run command to setup ${Logger.colors.cyan('npx nightwatch-mobile-helper android')}`,
+    `Run command for more help ${Logger.colors.cyan('npx nightwatch-mobile-helper android --help \n')}`
+  ];
+
+  err.stack = false;
+
+  throw err;
+};
+
+module.exports = class MobileServer {
+
+  constructor (AVD) {
+    this.sdkRoot = getSdkRootFromEnv(),
+    this.platform = getPlatformName();
+    this.avd = AVD || 'nightwatch-android-11';
+    this.emulatorId = '';
+    this.emulatorAlreadyRunning = true;
+  }
+
+  async killEmulator() {
+    if (this.emulatorAlreadyRunning) {
+      // Not killing the emulator when already running
+      process.exit(55);
+    }
+
+    return killEmulatorWithoutWait(this.sdkRoot, this.platform, this.emulatorId);
+  }
+  
+  async launchEmulator() {
+    return getAlreadyRunningAvd(this.sdkRoot, this.platform, this.avd).then(async (emulatorAlreadyRunning) => {
```

Comment:
`getAlreadyRunningAvd` also returns `emulatorId` or null if no emulator is found running. So, we should put it in `this.emulatorId` if an emulator is found running.